torch/distributed/_sharded_tensor/api.py

```diff
@@ -214,6 +223,76 @@ def _init_rpc(self):
         # Barrier for all RPCs to finish on all ranks.
         rpc.api._barrier(rpc_workers)
 
+    @classmethod
+    def init_from_local_shards(
+        cls,
+        local_shards: List[torch.Tensor],
+        shards_metadata: List[ShardMetadata],
+        process_group=None
+    ):
+
+        if len(local_shards) == 0 or len(shards_metadata) == 0:
+            raise ValueError("local_shards or shards_metadata must not be empty!")
```

Comment:
@pritamdamania I added the check here to check if `local_shards` is empty for each rank to call this. Do you think this is a valid check, or are there cases that in some rank `local_shards` can be empty? (i.e. shard_metadata.shard_lengths = [0, 0, ..] for the shard metadata in that rank)